unyt/dask_array.py

```diff
@@ -0,0 +1,620 @@
+"""
+a dask array class (unyt_dask_array) and helper functions for unyt.
+
+
+
+"""
+
+from functools import wraps
+
+import numpy as np
+
+import unyt.array as ua
+from unyt._on_demand_imports import _dask as dask
+
+__doctest_requires__ = {
+    ("dask_array.unyt_from_dask"): ["dask"],
+}
+
+
+_dask_Array = dask.array.core.Array
+_dask_finalize = dask.array.core.finalize
+
+# the following attributes hang off of dask.array.core.Array and do not modify units
+_use_simple_decorator = [
+    "min",
+    "max",
+    "sum",
+    "mean",
+    "std",
+    "cumsum",
+    "squeeze",
+    "rechunk",
+    "clip",
+    "view",
+    "swapaxes",
+    "round",
+    "copy",
+    "repeat",
+    "astype",
+    "reshape",
+    "topk",
+]
+
+
+def _simple_dask_decorator(dask_func, current_unyt_dask):
+    """
+    a decorator for the simpler dask functions that can just copy over the current
+    unit info after applying the dask function. this includes functions that return
+    single values (e.g., min()) or functions that change the array in ways that do
+    not affect units like reshaping or rounding.
+
+    Parameters
+    ----------
+
+    dask_func: func handle
+       the dask function handle to call
+    current_unyt_dask: unyt_dask_array
+       the current instance of a unyt_dask_array
+
+    Returns a new unyt_dask_array instance with appropriate units
+    """
+
+    @wraps(dask_func)
+    def wrapper(*args, **kwargs):
+        da = dask_func(*args, **kwargs)  # will return standard dask array
+        return _create_with_quantity(da, current_unyt_dask._unyt_array)
+
+    return wrapper
+
+
+# the following list are the unyt_quantity/array attributes that will also
+# be attributes of the unyt_dask_array. These methods are wrapped with a unit
+# conversion decorator.
+_unyt_funcs_to_track = ["to", "in_units", "in_cgs", "in_base", "in_mks"]
+
+
+def _track_conversion(unyt_func_name, current_unyt_dask):
```

Comment:
the way this function is used suggests it should be a method

Reply:
now a method! I think it was only a function because of how development evolved... but in its current form it does make more sense as a class method. which is now is.